pymc3/tests/test_distributions.py

```diff
@@ -751,25 +738,30 @@ def check_logcdf(
         if not skip_paramdomain_inside_edge_test:
             domains = paramdomains.copy()
             domains["value"] = domain
+
+            model, param_vars = build_model(pymc3_dist, domain, paramdomains)
+            pymc3_logcdf = model.fastfn(logpt(model["value"], cdf=True))
+
             if decimal is None:
                 decimal = select_by_precision(float64=6, float32=3)
 
             for pt in product(domains, n_samples=n_samples):
                 params = dict(pt)
-                if skip_params_fn(params):
-                    continue
-                scipy_cdf = scipy_logcdf(**params)
+                scipy_eval = scipy_logcdf(**params)
+
                 value = params.pop("value")
-                with Model() as m:
-                    dist = pymc3_dist("y", **params)
+                # Update shared parameter variables in pymc3_logcdf function
+                for param_name, param_value in params.items():
+                    param_vars[param_name].set_value(param_value)
+                pymc3_eval = pymc3_logcdf({"value": value})
+
                 params["value"] = value  # for displaying in err_msg
-                with aesara.config.change_flags(on_opt_error="raise", mode=Mode("py")):
```

Comment:
Are we setting `on_opt_error="raise"` elsewhere?  This is a good option to have set globally.

Also, are we sure we want to compile all these tests via C?  The Python-based graphs can sometimes be about as quick as the C-compiled graphs + their compilation time.

Reply:
> Are we setting on_opt_error="raise" elsewhere? This is a good option to have set globally.

I don't think we are. You mean set it up for all the tests?

> Also, are we sure we want to compile all these tests via C? The Python-based graphs can sometimes be about as quick as the C-compiled graphs + their compilation time.

I would say it's better to test them with the default back-end, which for the time being is `C`